src/executors/mod.rs

```diff
@@ -1,56 +1,56 @@
 /// This is the module that has all the executor functions
 /// i.e. the functions that have the responsibility of parsing and executing functions.
-use crate::types::{FunctionInfo, Request, Response};
-
-use std::collections::HashMap;
 use std::sync::Arc;
 
 use anyhow::{Context, Result};
 use log::debug;
-use pyo3_asyncio::TaskLocals;
-// pyO3 module
 use pyo3::prelude::*;
+use pyo3_asyncio::TaskLocals;
 
-fn get_function_output<'a>(
+use crate::types::{function_info::FunctionInfo, request::Request, response::Response};
+
+fn get_function_output<'a, T>(
     function: &'a FunctionInfo,
     py: Python<'a>,
-    request: &Request,
-) -> Result<&'a PyAny, PyErr> {
+    input: &T,
+) -> Result<&'a PyAny, PyErr>
+where
+    T: ToPyObject,
+{
     let handler = function.handler.as_ref(py);
-    let request_hashmap = request.to_hashmap(py).unwrap();
 
     // this makes the request object accessible across every route
     match function.number_of_params {
         0 => handler.call0(),
-        1 => handler.call1((request_hashmap,)),
+        1 => handler.call1((input.to_object(py),)),
         // this is done to accommodate any future params
-        2_u8..=u8::MAX => handler.call1((request_hashmap,)),
+        2_u8..=u8::MAX => handler.call1((input.to_object(py),)),
     }
 }
 
-pub async fn execute_middleware_function<'a>(
-    request: &Request,
-    function: FunctionInfo,
-) -> Result<HashMap<String, HashMap<String, String>>> {
-    // TODO:
-    // add body in middlewares too
-
+pub async fn execute_middleware_function<T>(input: &T, function: FunctionInfo) -> Result<T>
+where
+    T: for<'a> FromPyObject<'a> + ToPyObject,
```

Comment:
what does `for` do here?

Reply:
This was a tricky bug to solve, I didn't know about this either. I asked about it here on the pyo3 repo -> https://github.com/PyO3/pyo3/discussions/3058, and found more info about the syntax here -> https://doc.rust-lang.org/nomicon/hrtb.html

Reply:
Oh, wow. This will solve a lot of lifetime issues I faced.